### PR TITLE
Promote existing E2E for pod eviction with toleration timeout to Conformance - Single Pod Node

### PR DESCRIPTION
**Pull request by globervinodhn**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This PR requests to promote existing E2E test case to conformance. The E2E test verifies that the pod with toleration is not evicted from the tainted node before toleration timeout and is evicted only after the timeout seconds elapses.

**Which issue(s) this PR fixes**: Request for existing E2E test for conformance promotion
**Special notes for your reviewer**:
1. The existing E2E test case takes around 145 seconds for successful execution.
2. The test is non-flaky and non-disruptive in nature.
3. The test verifies that the tainted node does not evict the pod before toleration timeout.

**Does this PR introduce a user-facing change?**: NONE

release-note-none

/area conformance

cc @mgdevstack @brahmaroutu

### CODE REVIEW

File: test/e2e/scheduling/taints.go [hunk 1]
```diff
@@ -230,11 +230,13 @@ var _ = SIGDescribe("NoExecuteTaintManager Single Pod [Serial]", func() {
...
-	// 3. See if pod won't get evicted before toleration time runs out
-	// 4. See if pod will get evicted after toleration time runs out
-	It("eventually evict pod with finite tolerations from tainted nodes", func() {
+	/*
+		Release : v1.15
+		Testname: Taint, Pod toleration to no-execute with toleration timeout
+		Description: Create a Pod with toleration time-out and wait for it to be scheduled onto a Node. Taint the concerned Node
+						with a no-execute taint and verify that the Pod id evicted only after Toleration Timeout is reached. The test MUST pass.
+	*/
+	framework.ConformanceIt("eventually evict pod with finite tolerations from tainted nodes", func() {
```
Comment (timothysc) [inline test/e2e/scheduling/taints.go]: Have you verified this on an environment where all nodes are already tainted?  

/cc @johnSchnake 
/assign @timothysc

Comment (globervinodhn) [reply test/e2e/scheduling/taints.go]: If I understand it right, you mean to check if the pod is scheduled or not in the first place when the nodes are already tainted? Do correct if wrong...

Comment (globervinodhn) [reply test/e2e/scheduling/taints.go]: The E2E test verifies the pod behaviour which is already scheduled onto a node. Here once the pod is scheduled, the node is tainted with "NoExecute" which means that the pods which cannot tolerate the taint are to be evicted out. This happens at the earliest however, the test here considers a scenario wherein the pod has "Toleration timeout" which ensure that the pod is thrown out of the node only after the timeout is expired and not immediately after the node is tainted.

Comment (globervinodhn) [reply test/e2e/scheduling/taints.go]: > Have you verified this on an environment where all nodes are already tainted?
> 
> /cc @johnSchnake
> /assign @timothysc

Under this scenario, if the nodes are already tainted and if the pod has no tolerations for the taint then the pod is not scheduled at all.

**Comment by bclau** [comment]
/test pull-kubernetes-e2e-gce-device-plugin-gpu

I'm actually wondering if the test shouldn't be tagged as ``[Serial]``, since it taints a node, which could disrupt the execution of other tests. And if so, there was once a discussion whether to have Serial tests as conformance or not (I don't remember the outcome of the discussion though).

FWIW, it works on Windows.

**Review by timothysc — APPROVED** [review]
/lgtm

**Comment by timothysc** [comment]
/lgtm cancel 

Please bundle the other tests into a single PR.  This overlaps with https://github.com/kubernetes/kubernetes/pull/76985 .  You should do them together and answer the questions related to the PR.

**Comment by globervinodhn** [comment]
/hold

Reason: Need to merge this PR with that in #76985

**Comment by globervinodhn** [comment]
The test case is now a part of #76985  and hence this PR will be closed.

**Pull request closed without merging**